pkg/proto/nbcontract/v1/tlsbootstrappingconfig.proto

```diff
@@ -1,12 +1,9 @@
 syntax = "proto3";
 package nbcontract.v1;
 import "pkg/proto/nbcontract/v1/featurestate.proto";
-
+ 
 message TLSBootstrappingConfig {
-    // discuss whether this should be featureState or bool
-    // understand how defaulting works for enum, because we want default to be false
-    FeatureState tls_bootstrapping_status = 1; // need to review the relationship between tls_bootstrapping_status and secure_tls_bootstrapping_status. For example, if they are multually exclusive, then we should use oneof
-    FeatureState secure_tls_bootstrapping_status = 2;
-    string tls_bootstrap_token = 3; // Only required until Secure TLS bootstrapping in place. Would use kubelet identity after that.
-    string custom_secure_tls_bootstrap_appserver_appid = 4; // will follow up with the feature owner to see if this should be optional
+    bool enable_secure_tls_bootstrapping = 1;
```

Comment:
so you confirmed with the owner (Cameron, right?) that we only need a single variable to specify the status. Correct? That means, if enable_secure_tls_bootstrapping == false, then it's the original "insecure" tls bootstrapping.
Another, is there any reason we cannot use FeatureState secure_tls_bootstrapping_status? We've changed most of the enable/disable variables to _status.

Reply:
What will happen if both tls_bootstrap_token has value and enable_secure_tls_bootstrapping is true? 
Or could you let me know the 4 combinations of tls_bootstrap_token = empty or not, and enable_secure_tls_bootstrapping == true or false?
I am a bit confused actually.

Reply:
Yes I confirmed with Cameron that we only need a single variable as long we are able to populate both variables, `ENABLE_TLS_BOOTSTRAPPING` and `ENABLE_SECURE_TLS_BOOTSTRAPPING` in `cse_cmd`. It is NOT implied however, that if `enable_secure_tls_bootstrapping == true`, that the "not secure" tls bootstrapping is used. It is possible that neither is used.  Therefore, the 4 combinations are:

bootstrap_token = "" && enable_secure_tls_bootstrapping = false   -->   no tls boostrapping
bootstrap_token != "" && enable_secure_tls_bootstrapping = false   -->   normal tls boostrapping
bootstrap_token != "" && enable_secure_tls_bootstrapping = true   -->   secure tls boostrapping
bootstrap_token = "" && enable_secure_tls_bootstrapping = true   -->   secure tls boostrapping

Reply:
Also as referring to my previous comment, the reason why `enable_secure_tls_bootstrapping == false` does not imply that normal tls bootstrapping is used is because in AB we currently calculate whether to use normal tls bootstrapping simply by checking whether the token == "". Therefore in the helper functions we can populate `ENABLE_TLS_BOOTSTRAPPING` simply by checking whether the token is empty or token. If both `ENABLE_TLS_BOOTSTRAPPING` and `ENABLE_SECURE_TLS_BOOTSTRAPPING` is true, that is possible and secure tls will be used but that is an implementation detail for cameron in AB.

Reply:
Thanks for the truth table. Hopefully Cameron documented it somewhere. (Did he?)
You are right. We should use bool instead of FeatureState, as the default value for bool will be false in protobuff.

Reply:
docs are coming soon (TM)

Reply:
but yes Lily is correct :)